### PR TITLE
Connected processing to loading the dataset.

### PR DESCRIPTION
**Pull request by mooniean**
added processing when transformations are present. 
need to add documentation to explain how it works. 

added random seed

Addresses 
- [x] #9 
- [x] #6

### CODE REVIEW

File: tests/test_disk_io.py [hunk 1]
```diff
@@ -127,3 +129,33 @@ def test_get_loader_training_fail():
...
+def test_processing_data_some_transforms():
+    test_loader = DiskDataLoader(
+        pipeline=DISK_PIPELINE,
+        classes=DISK_CLASSES_FULL,
+        dataset_size=DATASET_SIZE_ALL,
+        training=True,
+        transformations=TRANSFORM_SOME,
+    )
+    test_loader.load(datapath=TEST_DATA_MRC, datatype=DATATYPE_MRC)
+    assert not test_loader.dataset.normalise
```
Comment (crangelsmith) [inline tests/test_disk_io.py:158]: we could add a test to make sure the transformation is happening, e.g. that the output dataset (for an example data point) is different if you pass a transformation to passing none.

File: tests/test_disk_io.py [hunk 2]
```diff
@@ -127,3 +129,33 @@ def test_get_loader_training_fail():
         torch_loader_train, torch_loader_val = test_loader.get_loader(
             split_size=1, batch_size=64
         )
+
```
Comment (crangelsmith) [inline tests/test_disk_io.py:132]: Also, it will be good to test the data loader returning what we want, I think here we have only tested that the class variables are correct (which is great!) but we won't catch things like the data loader not returning the right size of data, or selecting correctly the labels.

Comment (mooniean) [reply tests/test_disk_io.py:132]: that's a solid point! I checked it but realised it was only on a local function, I'll make that into assertions here

File: tests/test_disk_io.py (hunk 1 above)
Comment (mooniean) [reply tests/test_disk_io.py:239]: Latest commit addresses this!! I hope that's ok, made a debug flag that doesn't shuffle the paths so we can test the processing is happening.

**Comment by crangelsmith** [comment]
Looks good @mooniean! Just don't forget to add docstrings to document variables, but that can be in another PR to not stop   progress.

**Review by crangelsmith — APPROVED** [review]
